Dagger error: generated module has different visibility to abstract class

Build log:
``` kotlin
DetailsViewModelModule.java:9: error: This module is public, but it includes non-public (or effectively non-public) modules. Either reduce the visibility of this module or make app.tivi.showdetails.AssistedInject_DetailsViewModelModule public.
public abstract class DetailsViewModelModule {
                ^
```

The module is just a plain Kotlin public class:
``` kotlin
@AssistedModule
@Module(includes = [AssistedInject_DetailsViewModelModule::class])
abstract class DetailsViewModelModule
```

The generated class is pkg-private so I can't see a way to make it work in Kotlin

--- a/assisted-inject-processor-dagger2/src/main/java/com/squareup/inject/assisted/dagger2/processor/AssistedInjectDagger2Processor.kt
+++ b/assisted-inject-processor-dagger2/src/main/java/com/squareup/inject/assisted/dagger2/processor/AssistedInjectDagger2Processor.kt
@@ -33,6 +33,7 @@ import javax.annotation.processing.Processor
 import javax.annotation.processing.RoundEnvironment
 import javax.lang.model.SourceVersion
 import javax.lang.model.element.Element
+import javax.lang.model.element.Modifier
 import javax.lang.model.element.TypeElement
 import javax.tools.Diagnostic.Kind.ERROR
 
@@ -92,7 +93,8 @@ class AssistedInjectDagger2Processor : AbstractProcessor() {
     val targetNameToFactoryNames = targetTypeToFactoryType
         .map { (target, factory) -> TypeName.get(target.asType()) to ClassName.get(factory) }
         .toMap()
-    return AssistedInjectionModule(moduleName, targetNameToFactoryNames)
+    val public = Modifier.PUBLIC in moduleType.modifiers
+    return AssistedInjectionModule(moduleName, public, targetNameToFactoryNames)
   }
 
   private fun writeInflationModule(

--- a/assisted-inject-processor-dagger2/src/main/java/com/squareup/inject/assisted/dagger2/processor/AssistedInjectionModule.kt
+++ b/assisted-inject-processor-dagger2/src/main/java/com/squareup/inject/assisted/dagger2/processor/AssistedInjectionModule.kt
@@ -8,12 +8,14 @@ import com.squareup.javapoet.TypeName
 import com.squareup.javapoet.TypeSpec
 import javax.lang.model.element.Modifier.ABSTRACT
 import javax.lang.model.element.Modifier.PRIVATE
+import javax.lang.model.element.Modifier.PUBLIC
 
 private val MODULE = ClassName.get("dagger", "Module")
 private val BINDS = ClassName.get("dagger", "Binds")
 
 data class AssistedInjectionModule(
   val moduleName: ClassName,
+  val public: Boolean,
   val targetNameToFactoryName: Map<TypeName, ClassName>
 ) {
   val generatedType = moduleName.assistedInjectModuleName()
@@ -22,6 +24,11 @@ data class AssistedInjectionModule(
     return TypeSpec.classBuilder(generatedType)
         .addAnnotation(MODULE)
         .addModifiers(ABSTRACT)
+        .apply {
+          if (public) {
+            addModifiers(PUBLIC)
+          }
+        }
         .addMethod(MethodSpec.constructorBuilder()
             .addModifiers(PRIVATE)
             .build())

--- a/assisted-inject-processor-dagger2/src/test/java/com/squareup/inject/assisted/processor/dagger2/AssistedInjectDagger2ProcessorTest.kt
+++ b/assisted-inject-processor-dagger2/src/test/java/com/squareup/inject/assisted/processor/dagger2/AssistedInjectDagger2ProcessorTest.kt
@@ -77,6 +77,59 @@ class AssistedInjectDagger2ProcessorTest {
         .generatesSources(expected)
   }
 
+  @Test fun public() {
+    val test = JavaFileObjects.forSourceString("test.Test", """
+      package test;
+
+      import com.squareup.inject.assisted.Assisted;
+      import com.squareup.inject.assisted.AssistedInject;
+
+      class Test {
+        @AssistedInject
+        Test(Long foo, @Assisted String bar) {}
+
+        @AssistedInject.Factory
+        interface Factory {}
+      }
+    """)
+    val testFactory = JavaFileObjects.forSourceString("test.Test_AssistedFactory", """
+      package test;
+
+      class Test_AssistedFactory implements Test.Factory {}
+    """)
+    val module = JavaFileObjects.forSourceString("test.TestModule", """
+      package test;
+
+      import com.squareup.inject.assisted.dagger2.AssistedModule;
+      import dagger.Module;
+
+      @AssistedModule
+      @Module(includes = AssistedInject_TestModule.class)
+      public abstract class TestModule {}
+    """)
+
+    val expected = JavaFileObjects.forSourceString("test.AssistedInject_TestModule", """
+      package test;
+
+      import dagger.Binds;
+      import dagger.Module;
+
+      @Module
+      public abstract class AssistedInject_TestModule {
+        private AssistedInject_TestModule() {}
+
+        @Binds abstract Test.Factory bind_test_Test(Test_AssistedFactory factory);
+      }
+    """)
+
+    assertAbout(javaSources())
+        .that(listOf(test, testFactory, module))
+        .processedWith(AssistedInjectDagger2Processor())
+        .compilesWithoutError()
+        .and()
+        .generatesSources(expected)
+  }
+
   @Test fun moduleWithoutModuleAnnotationFails() {
     val moduleOne = JavaFileObjects.forSourceString("test.OneModule", """
       package test;

--- a/inflation-inject-processor/src/main/java/com/squareup/inject/inflation/processor/InflationInjectProcessor.kt
+++ b/inflation-inject-processor/src/main/java/com/squareup/inject/inflation/processor/InflationInjectProcessor.kt
@@ -26,6 +26,7 @@ import javax.lang.model.element.Element
 import javax.lang.model.element.ElementKind.CLASS
 import javax.lang.model.element.ElementKind.CONSTRUCTOR
 import javax.lang.model.element.ExecutableElement
+import javax.lang.model.element.Modifier
 import javax.lang.model.element.Modifier.PRIVATE
 import javax.lang.model.element.Modifier.STATIC
 import javax.lang.model.element.TypeElement
@@ -194,7 +195,8 @@ class InflationInjectProcessor : AbstractProcessor() {
   private fun InflationModuleElements.toInflationInjectionModule(): InflationInjectionModule {
     val moduleName = ClassName.get(moduleType)
     val inflationNames = inflationTypes.map { TypeName.get(it.asType()) }
-    return InflationInjectionModule(moduleName, inflationNames)
+    val public = Modifier.PUBLIC in moduleType.modifiers
+    return InflationInjectionModule(moduleName, public, inflationNames)
   }
 
   private fun writeInflationModule(

--- a/inflation-inject-processor/src/main/java/com/squareup/inject/inflation/processor/InflationInjectionModule.kt
+++ b/inflation-inject-processor/src/main/java/com/squareup/inject/inflation/processor/InflationInjectionModule.kt
@@ -20,6 +20,7 @@ private val STRING_KEY = ClassName.get("dagger.multibindings", "StringKey")
 
 data class InflationInjectionModule(
   val moduleName: ClassName,
+  val public: Boolean,
   val injectedNames: List<TypeName>
 ) {
   val generatedType = moduleName.inflationInjectModuleName()
@@ -27,7 +28,12 @@ data class InflationInjectionModule(
   fun brewJava(): TypeSpec {
     return TypeSpec.classBuilder(generatedType)
         .addAnnotation(MODULE)
-        .addModifiers(PUBLIC, ABSTRACT)
+        .addModifiers(ABSTRACT)
+        .apply {
+          if (public) {
+            addModifiers(PUBLIC)
+          }
+        }
         .addMethod(MethodSpec.constructorBuilder()
             .addModifiers(PRIVATE)
             .build())

--- a/inflation-inject-processor/src/test/java/com/squareup/inject/inflation/processor/InflationInjectProcessorTest.kt
+++ b/inflation-inject-processor/src/test/java/com/squareup/inject/inflation/processor/InflationInjectProcessorTest.kt
@@ -70,7 +70,7 @@ class InflationInjectProcessorTest {
       import dagger.multibindings.StringKey;
 
       @Module
-      public abstract class InflationInject_TestModule {
+      abstract class InflationInject_TestModule {
         private InflationInject_TestModule() {}
 
         @Binds
@@ -86,6 +86,62 @@ class InflationInjectProcessorTest {
         .compilesWithoutError()
         .and()
         .generatesSources(expectedFactory, expectedModule)
+  }
+
+  @Test fun public() {
+    val inputView = JavaFileObjects.forSourceString("test.TestView", """
+      package test;
+
+      import android.content.Context;
+      import android.util.AttributeSet;
+      import android.view.View;
+      import com.squareup.inject.assisted.Assisted;
+      import com.squareup.inject.inflation.InflationInject;
+
+      class TestView extends View {
+        @InflationInject
+        TestView(@Assisted Context context, @Assisted AttributeSet attrs, Long foo) {
+          super(context, attrs);
+        }
+      }
+    """)
+    val inputModule = JavaFileObjects.forSourceString("test.TestModule", """
+      package test;
+
+      import com.squareup.inject.inflation.InflationModule;
+      import dagger.Module;
+
+      @InflationModule
+      @Module(includes = InflationInject_TestModule.class)
+      public abstract class TestModule {}
+    """)
+
+    val expectedModule = JavaFileObjects.forSourceString("test.InflationModule_TestModule", """
+      package test;
+
+      import com.squareup.inject.inflation.ViewFactory;
+      import dagger.Binds;
+      import dagger.Module;
+      import dagger.multibindings.IntoMap;
+      import dagger.multibindings.StringKey;
+
+      @Module
+      public abstract class InflationInject_TestModule {
+        private InflationInject_TestModule() {}
+
+        @Binds
+        @IntoMap
+        @StringKey("test.TestView")
+        abstract ViewFactory bind_test_TestView(TestView_AssistedFactory factory);
+      }
+    """)
+
+    assertAbout(javaSources())
+        .that(listOf(inputView, inputModule))
+        .processedWith(InflationInjectProcessor())
+        .compilesWithoutError()
+        .and()
+        .generatesSources(expectedModule)
   }
 
   @Test fun assistedParametersLast() {

